### ISSUE TITLE
Keyboard navigation doesn't work correctly in Datepicker month and year views

### ISSUE THREAD

**Issue by ac-shadow** [chakra-ui/ark#3843]
### Description

The keyboard navigation doesn't work correctly in the month and year views of the datepicker. The focus seems to get stuck on the day view instead. Navigating while one of the other views is active just moves the focus in the hidden day view.

It seems like the `DatePicker.Table` doesn't get the correct view from the context. 
This can be seen in the basic stackblitz example when inspecting the rendered elements. All three tables have the data attributes for the day view.

This issue seems to only affect the vue version. 

Rendered elements (vue):

<img width="911" height="305" alt="Image" src="https://github.com/user-attachments/assets/20c41211-eb4f-472a-94e7-a947033fb450" />

Rendered elements (react):

<img width="936" height="305" alt="Image" src="https://github.com/user-attachments/assets/ace8ccaa-065f-457a-b3e0-73e6d5214d89" />

### Link to Reproduction (or Detailed Explanation)

https://stackblitz.com/edit/5ep6dknw?file=src%2FApp.vue

### Steps to Reproduce

1. Open the datepicker and switch to the month or year view
2. Try to navigate via keyboard

### Ark UI Version

5.35.0

### Framework

- [ ] React
- [ ] Solid
- [x] Vue
- [ ] Svelte

**Comment by segunadebayo** [comment]
This is a good catch @ac-shadow. I just pushed a fix for this. We'll release an update shortly.

If the issue persists after upgrading, I'll re-open it.
